### PR TITLE
Implement semantic versioning in Common lib

### PR DESCRIPTION
**Pull request by yahya130**
This PR includes the partial work to issues 56, the intention of this work is to implement a mutual semantic version scheme for all the apps using git tags

There are a couple of things I intended to tick off but did not due to my capacity:
1. validating new version tags are not less then current version when tagging and pushing (step exists but does not work properly)
2. end to end testing
3. checking if the pipeline could be moved into an action/reusable workflow 
4. check if there is a way to force annotated comments which are better than lightweight tagging

### CODE REVIEW

**Review by martinkearn — APPROVED** [review]

File: .github/workflows/publish-nuget-main.yml [hunk 1]
```diff
@@ -1,88 +1,105 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 6.0.x
+
+      - name: Install GitVersion
+        uses: gittools/actions/gitversion/setup@v0.9.13
```
Comment (keijoraamat) [inline .github/workflows/publish-nuget-main.yml:27]:
```suggestion
        uses: gittools/actions/gitversion/setup@9e6a3f8ae3192bb0f9750a10aeb5e1d4e3cf9055
```

File: .github/workflows/publish-nuget-main.yml [hunk 2]
```diff
@@ -1,88 +1,105 @@
...
+        with:
+          versionSpec: 5.x
+
+      - name: Determine Version
+        id: gitversion
+        uses: gittools/actions/gitversion/execute@v0.9.13
```
Comment (keijoraamat) [inline .github/workflows/publish-nuget-main.yml:33]:
```suggestion
        uses: gittools/actions/gitversion/execute@9e6a3f8ae3192bb0f9750a10aeb5e1d4e3cf9055
```

**Review by keijoraamat — APPROVED** [review]

File: .github/workflows/publish-nuget-main.yml (hunk 1 above)
Comment (decodingahmed) [reply .github/workflows/publish-nuget-main.yml:27]: Fair point, thank you!

File: .github/workflows/publish-nuget-main.yml (hunk 2 above)
Comment (decodingahmed) [reply .github/workflows/publish-nuget-main.yml:33]: Fair point, thank you!

**Pull request merged**